frontend/src/App.tsx

```diff
@@ -97,17 +98,28 @@ export default class App extends React.Component<{}, State> {
 
     this.state.nodes.setComparator(this.getComparator(this.sortBy.get()));
 
-    this.connection = Connection.create(this.pins, (changes) => {
-      if (changes) {
-        this.setState(changes);
-      }
-
-      return this.state;
-    });
+    this.connection = Connection.create(this.pins, this.setStateLazy);
 
     setInterval(() => (this.chainsCache = []), 10000); // Wipe sorted chains cache every 10 seconds
   }
 
+  private setStateLazy: Update = (changes) => {
+    // Apply new changes to the state immediately
+    Object.assign(this.state, changes);
```

Comment:
Yeah, I think at this point I can do away with using the React state entirely, and just make my own thin wrapper around state that triggers `forceUpdate` instead of `setState({})` at the end.